### PR TITLE
Add email

### LINKED ISSUE #62: Add email to contact
As an NUS computing student, I wish to add an email to each of my contacts, so that I have alternative means of contacting them when needed.

### PR DESCRIPTION
**Pull request by nknguyenhc**
Note:
* If the person already has the email, the programme simply ignores adding that email.
* If the user keys in same email in one command more than once, 
the programme gives an error message.

### CODE REVIEW

File: src/main/java/networkbook/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -96,6 +100,37 @@ public static Email parseEmail(String email) throws ParseException {
...
+    /**
+     * Parses a {@code Collection<String>} of emails into {@code UniqueList<Email>}.
+     * Leading and trailing whitespaces will be trimmed.
+     * @throws ParseException if at least one email in {@code Collection<String>} is invalid.
+     */
+    public static UniqueList<Email> parseEmails(Collection<String> emails) throws ParseException {
+        requireNonNull(emails);
+
+        if (!verifyNoDuplicates(emails)) {
+            throw new ParseException(MESSAGE_EMAIL_DUPLICATE);
```
Comment (awhb) [inline src/main/java/networkbook/logic/parser/ParserUtil.java:160]: Do add a test case with duplicate emails to check if the ParseException is thrown!

File: src/main/java/networkbook/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -96,6 +100,37 @@ public static Email parseEmail(String email) throws ParseException {
...
+        }
+        return result;
+    }
+
+    private static boolean verifyNoDuplicates(Collection<String> strings) {
+        Object[] stringArr = strings.toArray();
+        for (int i = 0; i < stringArr.length; i++) {
+            for (int j = i + 1; j < stringArr.length; j++) {
+                if (stringArr[i].equals(stringArr[j])) {
+                    return false;
```
Comment (awhb) [inline src/main/java/networkbook/logic/parser/ParserUtil.java:175]: Do test the verifyNoDuplicates function on a Collection<String> instance containing duplicate elements!

File: src/test/java/networkbook/logic/parser/ParserUtilTest.java [hunk 3]
```diff
@@ -150,6 +152,29 @@ public void parseEmail_validValueWithWhitespace_returnsTrimmedEmail() throws Exc
...
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseEmails(null));
+    }
+
+    @Test
+    public void parseEmails_collectionWithInvalidEmails_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseEmails(Arrays.asList(VALID_EMAIL, INVALID_EMAIL)));
+    }
+
+    @Test
+    public void parseEmails_emptyCollection_returnsEmptyListOfEmails() throws Exception {
```
Comment (awhb) [inline src/test/java/networkbook/logic/parser/ParserUtilTest.java:193]: I don't think we are supposed to let this function throw an Exception?

File: src/test/java/networkbook/logic/parser/ParserUtilTest.java [hunk 4]
```diff
@@ -150,6 +152,29 @@ public void parseEmail_validValueWithWhitespace_returnsTrimmedEmail() throws Exc
...
+        assertEquals(new UniqueList<Email>(), ParserUtil.parseEmails(Arrays.asList()));
+    }
+
+    @Test
+    public void parseEmails_collectionWithValidEmails_returnsEmailList() throws Exception {
```
Comment (awhb) [inline src/test/java/networkbook/logic/parser/ParserUtilTest.java:198]: Same issue as above, but do clarify if I've mistook something!

**Review by awhb — CHANGES_REQUESTED** [review]
In general looks good! But a few nits to take note of in terms of writing test cases. Will have to  trouble you to merge my PR locally too before you re-submit this PR!

File: src/test/java/networkbook/logic/parser/ParserUtilTest.java (hunk 3 above)
Comment (nknguyenhc) [reply src/test/java/networkbook/logic/parser/ParserUtilTest.java:193]: Please take a look at how other test cases are constructed for `ParserUtilTest`. They all have to throw exception in order for Java to compile. It is the same thing as writing a try-catch block, but if the method throws an exception, the test case is automatically failed.

File: src/main/java/networkbook/logic/parser/ParserUtil.java (hunk 1 above)
Comment (nknguyenhc) [reply src/main/java/networkbook/logic/parser/ParserUtil.java:160]: Done!

File: src/main/java/networkbook/logic/parser/ParserUtil.java (hunk 2 above)
Comment (nknguyenhc) [reply src/main/java/networkbook/logic/parser/ParserUtil.java:175]: Done!

**Review by awhb — APPROVED** [review]
LGTM!
